x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/analytics_list.tsx

```diff
@@ -119,17 +126,20 @@ export const DataFrameAnalyticsList: FC<Props> = ({
     isManagementTable
   );
 
-  const updateFilteredItems = (queryClauses: any) => {
-    if (queryClauses.length) {
-      const filtered = filterAnalytics(analytics, queryClauses);
-      setFilteredAnalytics(filtered);
-    } else {
-      setFilteredAnalytics(analytics);
-    }
-  };
+  const updateFilteredItems = useCallback(
```

Comment:
Now that you are able to restore the filter from the URL, could you improve the links from the Management page jobs list to pass e.g `id:bank_classification_1` rather than just `bank_classification_1` as happens currently and which may in theory not lead to exact matches? Same applies for the links to anomaly detection jobs.

```
http://localhost:5601/vdw/app/ml/data_frame_analytics?_a=(data_frame_analytics:(queryText:bank_classification_1))
```

Reply:
Done in https://github.com/elastic/kibana/pull/83439/commits/aa1f66bd1186587a398490a0ac4f1cced372f019